File: python/leccion11/Orden.py
from Producto import *


class Orden:
    contador_ordenes = 0

    def __init__(self, productos):
        Orden.contador_ordenes += 1
        self._id_orden = Orden.contador_ordenes
        self._productos = list(productos)

    def agregar_producto(self, producto):
        self._productos.append(producto)  # Esto es para agregar un nuevo producto

    def calcular_total(self):
        total = 0  # variable temporal para almacenar el total temporal
        for producto in self._productos:
            total += producto.precio
        return total

    def __str__(self):
        productos_str = ""
        for producto in self._productos:
            productos_str += producto.__str__() + "|"
        return f"Orden: {self._id_orden}, \nProducto: {productos_str}"


if __name__ == "__main__":
    producto1 = Producto("Camiseta", 100.00)
    producto2 = Producto("Pantalón", 150.00)
    producto1 = [producto1, producto2]
    orden1 = Orden(producto1)
    orden2 = Orden(producto1)
    print(orden1)
    print(orden2)

# Tarea: Modificar la orden2, ingresando nuevos productos con sus nombres y precios
# Crear una nueva lista de productos y agregarla a la orden2

producto3 = Producto("Zapatillas", 124.00)
producto4 = Producto("Medias", 20.00)
producto1 = [producto3, producto4]
orden2 = Orden(producto1)
print(orden2)

File: java/Leccion11/Ventas/src/ar/com/powersystem/ventas/Orden.java
package ar.com.powersystem.ventas;

public class Orden {

    private int idOrden;
    private Producto productos[];
    public static int contadorOrdenes;
    private int contadorProductos;
    static final int MAX_PRODUCTOS = 10;

    public Orden() {
        this.idOrden = ++Orden.contadorOrdenes;
        this.productos = new Producto[Orden.MAX_PRODUCTOS];
    }

    public void agregarProducto(Producto producto) {
        if (this.contadorProductos < Orden.MAX_PRODUCTOS) {
            this.productos[this.contadorProductos++] = producto;
        } else {
            System.out.println("Se ha superado el máximo de productos: " + Orden.MAX_PRODUCTOS);
        }
    }

    public double calcularTotal() {
        double total = 0;
        for (int i = 0; i < this.contadorProductos; i++) {
//            Producto producto = this.productos[i];
//            total += producto.getPrecio();
            total += this.productos[i].getPrecio();
        }
        return total;
    }

    public void mostrarOrden() {
        System.out.println("Id Orden: " + this.idOrden);
        double totalOrden = this.calcularTotal();
        System.out.println("El total en la orden es: $" + totalOrden);
        System.out.println("Productos de la orden: ");
        for (int i = 0; i < this.contadorProductos; i++) {
            System.out.println(this.productos[i]);
        }
    }
}
